### PR TITLE
Add E2E tests with kuttl

### PR DESCRIPTION
**Pull request by ccremer**
## Summary

* This PR adds e2e tests using Kuttl (https://kuttl.dev), an e2e testing tool for Kubernetes
* The tests do the following:
  - Create an objectsuser (+ assert conditions)
  - Create a Bucket (+ assert conditions)
  - Delete both (+ assert successful removal)

* Furthermore, the e2e tests are run in the master branch using a real-life API token for cloudscale.ch. To limit potential secret leak in forks or other PRs by running unreviewed code, we only run the e2e tests in master.
* The PR template adds a checklist for authors to encourage a successful e2e test run locally.

## Checklist

- [x] PR contains a single logical change (to keep release notes relevant)
- [x] Categorize the PR by setting a good title and adding one of the labels:
  `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
  as they show up in the changelog
- [ ] Link this PR to related issues

### CODE REVIEW

**Review by ZuGao — APPROVED** [review]
LGTM. Tests work.

File: test/local.mk [hunk 1]
```diff
@@ -101,3 +106,25 @@ $(webhook_key):
...
+# We need to set the Go arch since the binary is meant for the user's OS.
+$(kuttl_bin): export GOOS = $(shell go env GOOS)
+$(kuttl_bin): export GOARCH = $(shell go env GOARCH)
+$(kuttl_bin):
+	@mkdir -p $(kind_dir)
+	cd test/e2e && go build -o $@ github.com/kudobuilder/kuttl/cmd/kubectl-kuttl
+
+test-e2e: export KUBECONFIG = $(KIND_KUBECONFIG)
+test-e2e: $(kuttl_bin) local-install provider-config ## E2E tests
+	kubectl create ns --save-config e2e-test --dry-run=client -o yaml | kubectl apply -f -
```
Comment (Kidswiss) [inline test/local.mk:127]: Is there a specific reason to create the namespace this way? This seems a bit overcomplicated.

Isn't `kubectl create ns e2e-test` simpler?

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Just one remark, the rest LGTM

File: test/local.mk (hunk 1 above)
Comment (ccremer) [reply test/local.mk:127]: After a test run, the namespace `e2e-test` remains.
When `kubectl create` is executed a second time, it fails with "namespace exists already" error. Thus I use `kubectl apply` so it's idempotent but I need the YAML manifest first ("--save-config" creates the kubectl annotation)

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**